package/MDAnalysis/lib/_cutil.pyx

```diff
@@ -49,6 +49,70 @@ ctypedef cset[int] intset
 ctypedef cmap[int, intset] intmap
 
 
+@cython.boundscheck(False) # turn off bounds-checking for entire function 
+@cython.wraparound(False)  # turn off negative index wrapping for entire function 
+cdef np.intp_t[::1] _inverse_unique_unsorted_array(np.intp_t[::1] full_arr,
```

Comment:
Since we're modifying mask in place, I think it's clearer to have this with a void return type.

Reply:
Ah yes, missed that.
I was originally generating mask in that function but generating it outside and editing it inplace was quicker for some reason.